blocks/model.py

```diff
@@ -190,7 +181,18 @@ def get_params(self):
         parameters that do not belong to any brick.
 
         """
-        return self.params
+        brick_parameter_names = {
+            v: k for k, v in Selector(
+                self.top_bricks).get_parameters().items()}
+        parameter_list = []
+        for parameter in VariableFilter(roles=[PARAMETER])(
+                self.shared_variables):
+            if parameter in brick_parameter_names:
+                parameter_list.append((brick_parameter_names[parameter],
+                                       parameter))
+            else:
+                parameter_list.append((parameter.name, parameter))
+        return OrderedDict(parameter_list)
```

Comment:
Well it's bottom up rather than top down. You select all bricks and return all of their parameters with hierarchical path names rather than look for parameter's brick in order to compute the hierarchical path name. But I agree the result is the same. :)